sys/fuchsia/sys.txt

```diff
@@ -14,6 +14,8 @@ syz_thread_self() zx_thread
 syz_vmar_root_self() zx_vmar
 syz_job_default() zx_job
 
+# TODO: for x86 define none; for ARM define 1u << [1 thru 14]
```

Comment:
Are there literal defines for these features? If yes, then creating flags with these literal constants should work.

Reply:
There are literal defines but they're only defined in ARM builds, and I couldn't figure out a way to do #ifdef -style things within the syscall description syntax.  If you can tell me how I'll happily update this change (or do it in a follow-up pull request).

Reply:
This is generally support and should work, but there may be one part missing for fuchsia.

What we do for linux is as follows. You can run syz-extract manually for single file/arch. But when you run make extract, it extracts constants for all arches. To do that, syz-extracts bootstraps build for each arch in a separate temp dir, and then uses that temp dir to extract constants for the arch:
https://github.com/google/syzkaller/blob/master/sys/syz-extract/linux.go#L59

You can see for example:
```
syzkaller$ grep KVM_ARM_TARGET_CORTEX_A53 sys/linux/*.const
sys/linux/kvm_386.const:# KVM_ARM_TARGET_CORTEX_A53 is not set
sys/linux/kvm_amd64.const:# KVM_ARM_TARGET_CORTEX_A53 is not set
sys/linux/kvm_arm64.const:KVM_ARM_TARGET_CORTEX_A53 = 4
sys/linux/kvm_arm.const:# KVM_ARM_TARGET_CORTEX_A53 is not set
sys/linux/kvm_ppc64le.const:# KVM_ARM_TARGET_CORTEX_A53 is not set
```

Any syscalls that use such "is not set" constants are automatically disabled. So these arm specific ioctl's are used only on arm.

Since fuchsia has per-arch build dir and syz-extract uses there dirs:
https://github.com/google/syzkaller/blob/master/sys/syz-extract/fuchsia.go#L30
maybe this all will work out-of-the-box too? What happens when you try to use such arm-specific constants?